vm/actor/tests/common/mod.rs

```diff
@@ -231,12 +234,29 @@ impl<'a, BS: BlockStore> MockRuntime<'a, BS> {
             exit_code,
         })
     }
+
+    #[allow(dead_code)]
+    pub fn expect_create_actor(&mut self, code_id: Cid, address: Address) {
+        let a = ExpectCreateActor { code_id, address };
+        self.expect_create_actor = Some(a);
+    }
+
+    #[allow(dead_code)]
+    pub fn set_caller(&mut self, code_id: Cid, address: Address) {
+        self.message = UnsignedMessage::builder()
+            .to(self.message.to().clone())
+            .from(address.clone())
+            .build()
+            .unwrap();
```

Comment:
So one last thing, sorry this is for sure not ideal, but you also have to make sure the value (value_received in the mockrt) is retained, so include building the `value` as well. Sorry I wasn't explicit on my last review, that variable should also not be managed in two different places